### PR TITLE
lazysql: init at 0.1.8

### PR DESCRIPTION
**Pull request by kanielrkirby**
## Description of changes

Adding the `lazysql` package.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by bbenno** [comment]
Result of `nixpkgs-review pr 303238` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>lazysql</li>
  </ul>
</details>

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -10191,6 +10191,12 @@
     githubId = 56224949;
     name = "Mia Kanashi";
   };
+  kanielrkirby = {
+    email = "kanielrkirby@runbox.com";
+    github = "kanielrkirby";
+    githubId = 77940607;
+    name = "Kaniel Kirby";
+  };
```
Comment (h7x4) [inline maintainers/maintainer-list.nix:10199]: Could you split this change into a separate commit with message `maintainers: add kanielrkirby`?

File: pkgs/by-name/la/lazysql/package.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  };
+
+  vendorHash = "sha256-tgD6qoCVC1ox15VPJWVvhe4yg3R81ktMuW2dsaU69rY=";
+
+  buildInputs = [
+    xorg.libX11
+  ];
+
+  meta = with lib; {
+    description = "A cross-platform TUI database management tool written in Go.";
```
Comment (h7x4) [inline pkgs/by-name/la/lazysql/package.nix:25]:
```suggestion
    description = "A cross-platform TUI database management tool written in Go";
```

**Comment by kanielrkirby** [comment]
Should be good @h7x4!

**Comment by h7x4** [comment]
Sorry, could you squash the last two prs into a single commit named `lazysql: init at 0.1.8`? There are some [commit guidelines in place](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions)

**Comment by kanielrkirby** [comment]
My bad there. It's been a bit since I've made this one and I completely forgot about the guidelines :sweat_smile: Should (hopefully) be good now.

File: pkgs/by-name/la/lazysql/package.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    owner = "jorgerojas26";
+    repo = "lazysql";
+    rev = "v${version}";
+    hash = "sha256-yPf9/SM4uET/I8FsDU1le9JgxELu0DR9k7mv8PnBwvQ=";
+  };
+
+  vendorHash = "sha256-tgD6qoCVC1ox15VPJWVvhe4yg3R81ktMuW2dsaU69rY=";
+
+  buildInputs = [
+    xorg.libX11
```
Comment (h7x4) [inline pkgs/by-name/la/lazysql/package.nix:21]: Just a quick question about this - I saw the upstream issue about this. Do you know whether the package unconditionally needs libX11, or if it's possible to build without it on macos?

Comment (kanielrkirby) [reply pkgs/by-name/la/lazysql/package.nix:21]: @h7x4 I pulled out my Mac (M1) to test it, and it looks like it doesn't need it, just needs `Cocoa`. I can push those as changes if you'd like, making `libX11` depend on `isLinux` and `Cocoa` depend on `isDarwin`, if that sounds about right?

I don't have the devices to `x86_64-darwin` or `aarch64-linux`, but it compiles and runs on both `x86_64-linux` and `aarch64-darwin`.

Comment (h7x4) [reply pkgs/by-name/la/lazysql/package.nix:21]: That sounds good to me. I think we could just rely on ofborg building it for those architectures, assuming that they work, and rather fix it later if anyone complains.

Comment (kanielrkirby) [reply pkgs/by-name/la/lazysql/package.nix]: Mac architecture should be added now @h7x4

Comment (h7x4) [reply pkgs/by-name/la/lazysql/package.nix]: Cool. Just FYI, [the aarch64-darwin ofborg builders are having having a hard time keeping up](https://ofborg.org/prometheus/graph?g0.expr=ofborg_queue_builder_waiting%7Barch!%3D%22aarch64-darwin-lowprior%22%2Carch!%3D%22x86_64-darwin-lowprior%22%7D%20or%20ofborg_queue_evaluator_waiting&g0.tab=0&g0.stacked=0&g0.show_exemplars=0&g0.range_input=3d), so it might take some time for it to get built. But everything else looks fine, so feel free to ping me for merge when ofborg is done if I don't see it by then.

Comment (kanielrkirby) [reply pkgs/by-name/la/lazysql/package.nix]: Interesting, I'll be sure to let you know if I notice it being done. Really appreciate your time!

Comment (kanielrkirby) [reply pkgs/by-name/la/lazysql/package.nix]: @h7x4 Looks like everything passed! (If I understand `ofborg` at least lol)
